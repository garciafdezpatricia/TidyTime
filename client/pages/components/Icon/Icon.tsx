interface Props {
    src: string,
    alt: string,
    className? : string,
}

export function Icon ({src, className="btn-icon", alt } : Props) {
    return (
        // eslint-disable-next-line @next/next/no-img-element
        <img 
            className={className}
            src={src}
            alt={alt}>
        </img>
    )
}

