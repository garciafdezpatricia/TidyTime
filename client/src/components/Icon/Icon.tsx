// Copyright 2024 Patricia García Fernández.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

interface Props {
    src: string,
    alt: string,
    className? : string,
}

export function Icon ({src, className="btn-icon", alt } : Props) {
    return (
        // eslint-disable-next-line @next/next/no-img-element
        <img 
            className={className}
            src={src}
            alt={alt}>
        </img>
    )
}

