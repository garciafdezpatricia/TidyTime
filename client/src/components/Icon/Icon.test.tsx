import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import {Icon} from '@/src/components/Icon/Icon';

describe('Icon', () => {
    it('renders without crashing', () => {
      const { container } = render(<Icon src={""} alt={""}/>);
      expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Icon src={""} alt={""}/>);
        expect(asFragment()).toMatchSnapshot();
      });
  });