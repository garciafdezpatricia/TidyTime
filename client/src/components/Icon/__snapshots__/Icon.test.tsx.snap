// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon matches snapshot 1`] = `
<DocumentFragment>
  <img
    alt=""
    class="btn-icon"
    src=""
  />
</DocumentFragment>
`;
